inst/include/cpp11/as.hpp

```diff
@@ -62,7 +75,7 @@ inline bool is_convertable_without_loss_to_integer(double value) {
   return std::modf(value, &int_part) == 0.0;
 }
 
-template <typename T>
+template <typename T, typename = enable_if_t<!is_smart_ptr<T>::value>>
```

Comment:
I think it'd be clearer if `!is_smart_ptr` were added to the condition of `enable_if_constructible_from_sexp` with a comment explaining that it is a gcc 4.8 workaround

Reply:
I went back and forth between the two options :-)